### PR TITLE
test: replace bluebird .throw calls

### PR DESCRIPTION
**Pull request by jedwards1211**
_Please make sure to review and check all of these items:_

- [ ] Does `npm run test` or `npm run test-DIALECT` pass with this change (including linting)?
- [ ] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [ ] Have you added new tests to prevent regressions?
- [ ] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?
- [ ] Did you update the typescript typings accordingly (if applicable)?
- [ ] Did you follow the commit message conventions explained in [CONTRIBUTING.md](https://github.com/sequelize/sequelize/blob/master/CONTRIBUTING.md)?

### CODE REVIEW

**Comment by sushantdhiman** [comment]
Linting failed it seems

**Comment by sushantdhiman** [comment]
```js
) [SQLITE Specific] DAOFactory

       with storage "test.sqlite"

         .findOne

           should make aliased attributes available:

     AssertionError: expected undefined to equal 'user'

      at Proxy.assertEqual (node_modules/chai/lib/chai/core/assertions.js:1026:12)

      at Proxy.methodWrapper (node_modules/chai/lib/chai/utils/addMethod.js:57:25)

      at doAsserterAsyncAndAddThen (node_modules/chai-as-promised/lib/chai-as-promised.js:289:22)

      at Proxy.<anonymous> (node_modules/chai-as-promised/lib/chai-as-promised.js:255:20)

      at Proxy.overwritingMethodWrapper (node_modules/chai/lib/chai/utils/overwriteMethod.js:78:33)

      at User.findOne.then.user (test/integration/dialects/sqlite/dao-factory.test.js:118:40)

  From previous event:

      at Context.<anonymous> (test/integration/dialects/sqlite/dao-factory.test.js:117:16)

      at AsyncResource.runInAsyncScope (async_hooks.js:188:21)

  From previous event:
```

**Comment by jedwards1211** [comment]
@sushantdhiman I noticed that test is now skipped in master, let me rebase...

**Comment by jedwards1211** [comment]
@sushantdhiman this test failure doesn't seem related to my changes

File: test/integration/dialects/sqlite/dao-factory.test.js [hunk 1]
```diff
@@ -110,7 +110,7 @@ if (dialect === 'sqlite') {
             });
           });
 
-          it.skip('should make aliased attributes available', function() {
+          it('should make aliased attributes available', function() {
```
Comment (sushantdhiman) [inline test/integration/dialects/sqlite/dao-factory.test.js:113]:
```suggestion
          it.skip('should make aliased attributes available', function() {
```

Comment (jedwards1211) [reply test/integration/dialects/sqlite/dao-factory.test.js:113]: weird, rebasing didn't work...

**Pull request merged**